schedule/models/events.py

```diff
@@ -59,6 +62,82 @@ def __unicode__(self):
     def get_absolute_url(self):
         return reverse('event', args=[self.id])
 
+    def get_recent_start_validated(self):
+        """
+         Unserializes and validates self.recent_start. Returns the dictionary
+         represented by self.recent_start if all goes well, or None if there
+         is a validation error.
+         @return recent_start converted to object if it's valid given current
+         state of event; None if it doesn't exist or is invalid.
+        """
+        if self.recent_start == None:
+            return None
+
+        recent_start_data = pickle.loads(str(self.recent_start))
+
+        if (recent_start_data['event']['start'] == self.start and
```

Comment:
We can alternatively add a post_save signal to nil out recent_start_data on writes, but this works without any performance issues as well.